app/src/angular/classes/wallet.js

```diff
@@ -254,11 +257,21 @@ class Wallet {
             if (contractDataHex) {
                 rawTx.data = EthUtils.sanitizeHex(contractDataHex);
             }
-
-            let eTx = new Tx(rawTx);
-            eTx.sign(this.privateKey);
-
-            defer.resolve('0x' + eTx.serialize().toString('hex'));
+            
+            if (!this.isLedger) {
```

Comment:
We should be using the factory pattern here to allow for adding other integrations in the future.